data-access-layer/iaas/CloudProviderClient.js

```diff
@@ -61,10 +61,10 @@ class CloudProviderClient extends BaseCloudClient {
   list(container, options) {
     if (arguments.length < 2) {
       options = container;
-      container = this.containerName;
+      container = _.get(options, 'container') || this.containerName;
     }
     return this.storage
-      .getFilesAsync(container, options)
+      .getFilesAsync(container, _.omit(options, 'container'))
```

Comment:
Then I guess, this will also be removed if we are not passing container in options.

Reply:
Yes. Removed these changes.